management-microservice/src/main/java/com/nearsoft/fgaribay/mgmt/microservice/ProductManagementService.java

```diff
@@ -0,0 +1,46 @@
+package com.nearsoft.fgaribay.mgmt.microservice;
+
+import com.nearsoft.fgaribay.mgmt.ProductRepository;
+import com.nearsoft.fgaribay.mgmt.model.Product;
+import com.nearsoft.fgaribay.mgmt.model.ProductRequest;
+import com.nearsoft.fgaribay.mgmt.model.ProductResponse;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+@Component
+public class ProductManagementService {
+
+  private ProductRepository productRepository;
+
+  @Autowired
+  public ProductManagementService(ProductRepository productRepository) {
+    this.productRepository = productRepository;
+  }
+
+  @RabbitListener(queues = "${product.queues.list-name}")
+  public ProductResponse listProducts(ProductRequest request) {
+
+    ArrayList<Product> products = new ArrayList<>(productRepository.getAllProducts());
+    boolean error = false;
+    String errorMessage = "";
+    UUID uuid = request.getId();
+
+    return new ProductResponse("list", error, errorMessage, products, uuid);
+  }
+
+  @RabbitListener(queues = "${product.queues.creation-name}")
+  public ProductResponse createProduct(ProductRequest request) {
+
+    boolean error = false;
+    String errorMessage = "";
+    UUID uuid = request.getId();
+    Product product = request.getProducts().get(0);
```

Comment:
Why are you sending a list if at the end you only read the first value?

Reply:
I use the same ProductRequest and ProductResponse for both current microservice operations (create one product and list all). Other operations might be required in the future such as creating multiple products. When creating only one product, it would be on the first position.

This is probably a code smell, but I am unsure whether the best solution would be to use a specific type of request for creating a single product that does not use a List or to have validation on the actual size of the List in the response and throw an error if it is not exactly one.

Do you have any suggestions? If not, I will go with a minimal change adding validation and discuss the preferred architecture with my mentors.

Reply:
Hi Ivan! Talking to my mentors, we decided that it might look better to use generics here, so instead of having a ProductRequest, we would have a ServiceRequest with a generic data field.

If it's ok with you, I would like to leave that for the next pull request, which I will try to submit today.